### ISSUE TITLE
STL problems

### ISSUE THREAD

**Issue by ginfung** [avanti534/CPP_Programming#5]
UVa 101
UVa 10815 (do this again)
UVa 156
UVa400
UVa10474

**Comment by avanti534** [comment]
Hi, I was reviewing over the question and got a little confused as to what I should to solve in UVa 156. Could you please explain what the program is meant to do?
Thank you

**Comment by ginfung** [comment]
@avanti534 UVa 156 asks for printing out all "ananagrams" in dict order.
Consider the sample input
```
ladder came tape soon leader acme RIDE lone Dreis peat
ScAlE orb eye Rides dealer NotE derail LaCeS drIed
noel dire Disk mace Rob dries
```
ladder should be one ananagram. 

came should not count as ananagram, since there exists any other word in the text -- acme (same letters with different order)

"tape" should not count, given "peat"

"Rides" should not count, given "dries" (same letters with a different order. case insensitive)

Feel free to use `std::set`, `std::map`, etc. 
`std::sort` is helpful too! (https://stackoverflow.com/questions/9107516/sorting-characters-of-a-c-string)

**Comment by ginfung** [comment]
https://www.geeksforgeeks.org/binary-search-functions-in-c-stl-binary_search-lower_bound-and-upper_bound/

**Comment by avanti534** [comment]
I have uploaded my current code for the Marbles project (UVA 10474), and am currently stuck facing an issue with the queries and marble number.
The code seems to be finding the wrong length of the marble numbers length from the input, which as a result incorrectly reads the next proper query.

I have uploaded two screenshots as well in the STL folder, the console screenshot shows what the output of the code reads, while input is the input. My guess from the output is that it incorrectly loops through the list to find the queries. I.E, 4 works fine because it is the first index, but after that it seems to be one short for the end of each marble list to the next query, with 5, and then the additional query of 3 at the end. 

In summary, I am having a problem with the code reading the queries and marbles correctly, how can I fix this?

Thank you!

**Comment by ginfung** [comment]
@avanti534 The result is correct. We have two problem sets. The first set has one query -- 5. The other one with two queries -- 2 and 3.
Therefore, our printing out should be "5 2 3", which is consistent with your screenshot.

You might read the problem statement again to better understand the input formats.
